dynamic-layers/openembedded-layer/recipes-devtools/bcc/bcc/0001-Vendor-just-enough-extra-headers-to-allow-libbpf-to-.patch

```diff
@@ -0,0 +1,124 @@
+From 6cffc195eca6b53a12865d325ff97e7b5ba8f22b Mon Sep 17 00:00:00 2001
+From: Daniel Thompson <daniel.thompson@linaro.org>
+Date: Thu, 19 May 2022 09:14:20 +0000
+Subject: [PATCH] Vendor just enough extra headers to allow libbpf to be
+ de-vendored
+
+Currently it is not possible to build the bcc recipe when we de-vendor
+libbpf and adopt the packaged version. Ironically this is due to the
+deliberate vendoring of some depreciated btf code that is being removed
+upstream because bcc was the only user! In other words the vendored code
+doesn't work the de-vendored libbpf because also ends up de-vendoring
+one of the Linux uapi headers.
+
+This is obviously an OE specific issue (due to the current combination
+of linux headers, libbpf and bcc). It's a bit of a mess and the right
+solution is probably to update the system UAPI headers but I am committing
+this for now simply so I can easily show why we must de-vendor libbpf in
+the first place!
```

Comment:
this patch has significant cost of maintenance in future if it  remains here, so I am a bit worried. Have you discussed this with upstream bcc community to see how they want to handle this situation for future ?

Reply:
I see, you mean linux UAPI headers right ? which we maybe doing soon,

Reply:
Yep, UAPI headers. If that gets updated then it should be easy to get bpftrace working on master by de-vendoring the bcc build.

I assume kirkstone will stay on the older UAPI though, right?

BTW and FYI, upstream discussion is here: https://github.com/iovisor/bcc/issues/4014

Reply:
Yes Kirk stone will stay where it is

Reply:
Hi @kraj,

Can you help backport this patch to kirkstone stable branch?

-Sumit